semgrep/semgrep/rule.py

```diff
@@ -65,7 +72,35 @@ def __init__(self, raw: YamlTree[YamlMap]) -> None:
         self._includes = path_dict.get("include", [])
         self._excludes = path_dict.get("exclude", [])
         self._languages = [Language(l) for l in self._raw["languages"]]
-        self._expression = self._build_boolean_expression(self._yaml)
+
+        # check taint/search mode
+        self._expression, self._mode = self._taint_or_search_patterns_validation(
+            self._yaml
+        )
+
+    def _taint_or_search_patterns_validation(
+        self, rule: YamlTree[YamlMap]
+    ) -> Tuple[BooleanRuleExpression, Mode]:
+
+        rule_raw = rule.value
+        _mode = rule_raw.get("mode")
```

Comment:
You can just use `rule_raw.get("mode", DEFAULT_MODE)` here, and remove a case below.

Reply:
It won't let me put more than 1 arg into get() because rule_raw is a YamlMap object, not a dictionary. Do you have another idea? This approach would be cleaner...

Reply:
It didn't like the "or" approach either, so I went with a ternary, which has a similar effect wrt code structure and complexity